File: java/src/main/java/fr/sdecout/handson/rest/shared/ValidIsbnValidator.java
package fr.sdecout.handson.rest.shared;

import jakarta.validation.ConstraintValidator;
import jakarta.validation.ConstraintValidatorContext;

public final class ValidIsbnValidator implements ConstraintValidator<ValidIsbn, String> {

    @Override
    public boolean isValid(String value, ConstraintValidatorContext constraintValidatorContext) {
        return value == null || Isbn.isValid(value);
    }

}

File: kotlin/src/main/kotlin/fr/sdecout/handson/rest/shared/ValidIsbnValidator.kt
package fr.sdecout.handson.rest.shared;

import jakarta.validation.ConstraintValidator
import jakarta.validation.ConstraintValidatorContext

class ValidIsbnValidator : ConstraintValidator<ValidIsbn, String?> {
    override fun isValid(value: String?, constraintValidatorContext: ConstraintValidatorContext): Boolean =
        value == null || value.replace("-", "").matches(Regex("[0-9]{13}"))
}
